### DISCUSSION TITLE
pure browser or need node enviroment.

### DISCUSSION THREAD

**Discussion by liuhang-cn** [xmppjs/xmpp.js#929] [Q&A] [answered]
Xmpp.js can run in pure browser or need node enviroment?
I try to use it in vue project, but some node enviroment is needed when i run
![20211220210650](https://user-images.githubusercontent.com/20272724/146771814-65ae132c-a50f-4125-ace5-a806e3a87b50.png)
.

**Comment by sonnyp** [comment, marked as answer]
How did you include it?

If you don't want to use rollup or similar - you'll have to use the bundle 

https://github.com/xmppjs/xmpp.js/tree/main/packages/client#setup

```
<script src="https://unpkg.com/@xmpp/client@0.13.0/dist/xmpp.min.js" crossorigin/>
```
